pkg/appgateserver/config/appgate_configs_reader_test.go

```diff
@@ -63,43 +67,85 @@ func Test_ParseAppGateConfigs(t *testing.T) {
 
 			inputConfig: ``,
 
-			expectedError: config.ErrAppGateConfigUnmarshalYAML,
+			expectedError: config.ErrAppGateConfigEmpty,
 		},
 		{
 			desc: "invalid: no signing key",
 
 			inputConfig: `
 				self_signing: true
-				signing_key:
+				// explicitly missing signing key
 				listening_endpoint: http://localhost:42069
-				query_node_url: tcp://127.0.0.1:36657
+				query_node_grpc_url: tcp://127.0.0.1:36658
+				query_node_rpc_url: tcp://127.0.0.1:36657
 				`,
 
-			expectedError: config.ErrAppGateConfigEmptySigningKey,
+			expectedError: config.ErrAppGateConfigUnmarshalYAML,
 		},
 		{
 			desc: "invalid: invalid listening endpoint",
 
 			inputConfig: `
 				self_signing: true
 				signing_key: app1
-				listening_endpoint: &localhost:42069
-				query_node_url: tcp://127.0.0.1:36657
+				listening_endpoint: l&ocalhost:42069
+				query_node_grpc_url: tcp://127.0.0.1:36658
+				query_node_rpc_url: tcp://127.0.0.1:36657
 				`,
 
 			expectedError: config.ErrAppGateConfigInvalidListeningEndpoint,
 		},
 		{
-			desc: "invalid: invalid query node url",
+			desc: "invalid: invalid query node grpc url",
+
+			inputConfig: `
```

Comment:
NIT: wdyt about renaming this to something like `inputConfigYAML` and/or adding a comment in the field declaration.